### PR TITLE
Fix enhanced remote context when multiple repos are added

### LINKED ISSUE #1081: bug: Adding multiple remote repos does not work as expected
### Installation Information

Any cody version with support for remote repos

### Describe the bug

When adding multiple repositories only context from last one is properly included

### Expected behavior

All remote repositories should be used

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by pkukielka**
Fixes https://github.com/sourcegraph/jetbrains/issues/1081

## Changes

Previously when adding multiple repositories last one added was effectively overriding al the previous ones.
That was leading to lacking context and surprising behaviour depending on the order of the repos added.

## Test plan

Manual tests according to QA testing guide but with multiple repositories added, and enabling/disabling single repositories.

### CODE REVIEW

File: src/main/kotlin/com/sourcegraph/cody/context/ui/AddRepositoryDialog.kt [hunk 1]
```diff
@@ -61,11 +61,10 @@ class AddRepositoryDialog(
...
               codebaseName ?: return@custom false
-              val repo =
-                  RemoteRepoUtils.getRepository(project, codebaseName)
-                      .completeOnTimeout(null, 2, TimeUnit.SECONDS)
-                      .get()
-              repo != null
+              RemoteRepoUtils.getRepositories(project, listOf(codebaseName))
+                  .completeOnTimeout(null, 2, TimeUnit.SECONDS)
+                  .get()
+                  .isNotEmpty()
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/context/ui/AddRepositoryDialog.kt]: nit: I guess it should be a singleton, right?

File: src/main/kotlin/com/sourcegraph/cody/context/ui/EnhancedContextPanel.kt [hunk 2]
```diff
@@ -92,10 +92,10 @@ class EnhancedContextPanel(private val project: Project, private val chatSession
       } else {
         CodyAgentCodebase.getInstance(project).getUrl().thenApply { repoUrl ->
           val codebaseName = convertGitCloneURLToCodebaseNameOrError(repoUrl)
-          RemoteRepoUtils.getRepository(project, codebaseName)
+          RemoteRepoUtils.getRepositories(project, listOf(codebaseName))
               .completeOnTimeout(null, 15, TimeUnit.SECONDS)
-              .thenApply { repo ->
-                if (repo != null) {
+              .thenApply { repos ->
+                if (repos.isNotEmpty()) {
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/context/ui/EnhancedContextPanel.kt]: nit: what if it's `null`?

**Review by mkondratek — APPROVED** [review]
tested on 2023.3 - works 🚀

File: src/main/kotlin/com/sourcegraph/cody/context/ui/EnhancedContextPanel.kt (hunk 2 above)
Comment (pkukielka) [reply src/main/kotlin/com/sourcegraph/cody/context/ui/EnhancedContextPanel.kt]: Fixed

File: src/main/kotlin/com/sourcegraph/cody/context/ui/AddRepositoryDialog.kt (hunk 1 above)
Comment (pkukielka) [reply src/main/kotlin/com/sourcegraph/cody/context/ui/AddRepositoryDialog.kt]: Changes

**Pull request merged**